Mathlib/RingTheory/DedekindDomain/Different.lean

```diff
@@ -392,6 +395,77 @@ lemma dual_injective :
     Function.Injective (dual A K : FractionalIdeal B⁰ L → FractionalIdeal B⁰ L) :=
   dual_involutive.injective
 
+variable (A K B L)
+
+omit [FiniteDimensional K L] in
+theorem dual_eq_dual_mul_dual (C M : Type*) [CommRing C] [IsDedekindDomain C] [Field M]
+    [Algebra C M] [IsFractionRing C M] [Algebra A C] [Algebra B C] [Algebra A M] [Algebra B M]
+    [Algebra K M] [Algebra L M] [IsScalarTower A C M] [IsScalarTower A K M] [IsScalarTower B C M]
+    [IsScalarTower B L M] [IsScalarTower K L M] [IsIntegralClosure C A M] [IsIntegralClosure C B M]
+    [NoZeroSMulDivisors B C] [FiniteDimensional K M] [Algebra.IsSeparable K M] :
+    haveI h : B⁰ ≤ Submonoid.comap (algebraMap B C) C⁰ :=
+      nonZeroDivisors_le_comap_nonZeroDivisors_of_injective _ <|
+        FaithfulSMul.algebraMap_injective _ _
+    haveI : Module.Finite L M := Module.Finite.right K L M
+    haveI : Module.Finite K L := Module.Finite.left K L M
+    haveI : Algebra.IsSeparable L M := isSeparable_tower_top_of_isSeparable K L M
+    dual A K (1 : FractionalIdeal C⁰ M) = dual B L (1 : FractionalIdeal C⁰ M) *
+        (dual A K (1 : FractionalIdeal B⁰ L)).extended M h := by
```

Comment:
I think typeclass needed in the statement should be arguments, even if they are implied by other typeclasses.
Because you will need the duplicated typeclasses in your context for such a goal to appear anyways.